xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoomManager.java

```diff
@@ -10,15 +10,25 @@
 
 /**
  * this class supports the simple LocalMUCRoom management including remove,add and query.
+ *
+ * Note that this implementation provides a representation of rooms that are currently actively loaded in memory only.
+ * More rooms might exist in the database.
```

Comment:
Is it worth explaining why a room may be in the DB but not in memory?

Reply:
> > I attempted to keep API changes to a minimum.
> 
> As I reviewed this, I don't think I found any API breaks. Is that actually the case?

I think so. I added a couple of new methods, which, where possible, got a default implementation, so I think we're safe.